app/assets/javascripts/select-kit/addon/components/bulk-select-bookmarks-dropdown.js

```diff
@@ -0,0 +1,113 @@
+import { service } from "@ember/service";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import Bookmark from "discourse/models/bookmark";
+import i18n from "discourse-common/helpers/i18n";
+import DropdownSelectBoxComponent from "select-kit/components/dropdown-select-box";
+
+const _customButtons = [];
+const _customActions = {};
+
+export function addBulkDropdownAction(name, customAction) {
+  _customActions[name] = customAction;
+}
+
+export function addBulkDropdownButton(opts) {
+  _customButtons.push({
+    id: opts.label,
+    icon: opts.icon,
+    name: i18n(opts.label),
+    visible: opts.visible,
+  });
+  addBulkDropdownAction(opts.label, opts.action);
+}
+
+export default DropdownSelectBoxComponent.extend({
+  classNames: ["bulk-select-bookmarks-dropdown"],
+  headerIcon: null,
+  showFullTitle: true,
+  selectKitOptions: {
+    showCaret: true,
+    showFullTitle: true,
+    none: "select_kit.components.bulk_select_bookmarks_dropdown.title",
+  },
+
+  router: service(),
+  toasts: service(),
+  dialog: service(),
+
+  computeContent() {
+    let options = [];
+    options = options.concat([
+      {
+        id: "clear-reminders",
+        icon: "tag",
+        name: i18n("bookmark_bulk_actions.clear_reminders.name"),
+      },
+      {
+        id: "delete-bookmarks",
+        icon: "trash-alt",
+        name: i18n("bookmark_bulk_actions.delete_bookmarks.name"),
+      },
+    ]);
+
+    return [...options, ..._customButtons];
+  },
+
+  getSelectedBookmarks() {
+    return this.bulkSelectHelper.selected;
+  },
+
+  @action
+  onSelect(id) {
+    switch (id) {
+      case "clear-reminders":
+        this.dialog.yesNoConfirm({
+          message: i18n(
+            `js.bookmark_bulk_actions.clear_reminders.description`,
+            {
+              count: this.getSelectedBookmarks().length,
+            }
+          ),
+          didConfirm: () => {
+            Bookmark.bulkOperation(this.getSelectedBookmarks(), {
+              type: "clear_reminder",
+            })
+              .then(() => {
+                this.router.refresh();
+                this.bulkSelectHelper.toggleBulkSelect();
+                this.toasts.success({
+                  duration: 3000,
+                  data: { message: i18n("bookmarks.bulk.reminders_cleared") },
+                });
+              })
+              .catch(popupAjaxError);
+          },
+        });
+        break;
+      case "delete-bookmarks":
+        this.dialog.deleteConfirm({
+          message: i18n(
+            `js.bookmark_bulk_actions.delete_bookmarks.description`,
+            {
+              count: this.getSelectedBookmarks().length,
+            }
+          ),
+          didConfirm: () => {
+            Bookmark.bulkOperation(this.getSelectedBookmarks(), {
+              type: "delete",
+            })
+              .then(() => {
+                this.router.refresh();
+                this.bulkSelectHelper.toggleBulkSelect();
```

Comment:
I'm not sure we need to toggle the bulk select state here. Maybe the user wants to continue bulk selecting and editing other bookmarks?

Reply:
It seems that the toggle is not necessary since the router refresh is already cleaning up the selection. 

In terms of user experience, however,  I can't see a specific case where keeping the selection would be helpful for the current "bookmark's bulk operations." 

If the user deletes the bookmark, he can't clean up the reminders, and if they spend effort cleaning the reminders, I don't think they will want to delete the same bookmarks in a subsequent operation. 

Are you thinking about a specific workflow? Maybe I'm missing something; I will be happy to fix this; just thought it could be something to think about and discuss :)

Reply:
Yes, if you have say 20 bookmarks, you might want to: 

- select 5, clear reminders 
- select 2, delete
- continue with more selections

Reply:
Ooh, sorry for the confusion! I misunderstood your suggestion. I thought you wanted to keep the checkboxes selected, which is not what you are suggesting 😅 
So, this is what we have  at this point - I guess it matches your initial idea:
<img width="1491" alt="Bookmarks - Activity - amandaalves15 - Discourse 2024-05-07 at 10 57 17 PM" src="https://github.com/discourse/discourse/assets/15239141/4578fb47-4a2c-4f8d-a1db-f1abcdd14b19">